Code/GraphMol/SmilesParse/SmilesWrite.cpp

```diff
@@ -330,11 +319,11 @@ std::string FragmentSmilesConstruct(
                               bondsInPlay, bondSymbols, doIsomericSmiles,
                               doRandom);
   Bond *bond = nullptr;
-  BOOST_FOREACH (Canon::MolStackElem mSE, molStack) {
+  for (auto &mSE : molStack) {
     switch (mSE.type) {
       case Canon::MOL_STACK_ATOM:
         if (!ringClosuresToErase.empty()) {
-          BOOST_FOREACH (unsigned int rclosure, ringClosuresToErase) {
+          for (auto rclosure : ringClosuresToErase) {
```

Comment:
The ringClosuresToErase.empty() call is not needed.

Reply:
correct